src/test/addrman_tests.cpp

```diff
@@ -897,7 +885,9 @@ BOOST_AUTO_TEST_CASE(addrman_evictionworks)
     BOOST_CHECK_EQUAL(info.ToString(), "250.1.1.19:0");
 
     // Ensure test of address fails, so that it is evicted.
-    addrman.SimConnFail(info);
+    // Set last good connection in the deep past.
+    addrman.Good(info, /*nTime=*/1);
```

Comment:
https://github.com/bitcoin/bitcoin/pull/23826/commits/02f5a618dc418da2d6e4af141774dc24a1c6a65d: nit (doesn't change the test behavior): I'd recommend checking the output from `addrman->Good()` whenever it is called.

1) it is consistent with how `Good()` is called in the rest of the tests
2) it improves the readability (are you expecting a true or false from `Good()`)

```suggestion
    BOOST_CHECK(!addrman.Good(info, /*nTime=*/1));
```

for example, while first reviewing this change I expected an output of `true` from `Good()`, but after adding `BOOST_CHECK` realized `Good()` was actually returning `false` because `info` is already in the tried set

Reply:
done, and I also changed the comment a bit to reflect that.